Sources/SpeziViews/Utilities/LocalPreferences/LocalPreference.swift

```diff
@@ -0,0 +1,229 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+// swiftlint:disable file_types_order
+
+import Foundation
+import SpeziFoundation
+import SwiftUI
+
+
+/// A type-safe alternative to SwiftUI's `AppStorage`.
+///
+///
+/// The `@LocalPreference` property wrapper accesses the key's corresponding value in the `UserDefaults`.
+/// This property wrapper is auto-updating.
+///
+/// By default, the ``LocalPreferencesStore/standard`` `UserDefaults` store is used, but you can customise this (see ``init(_:store:)``).
+///
+/// Similar to `@State` and `@AppStorage`, you can use the `$`-prefix notation to obtain a `Binding` to the value.
+///
+/// ```swift
+/// enum RootTab: String {
+///     case home, feed, account
+/// }
+///
+/// extension LocalPreferenceKeys {
+///     static let rootTab = LocalPreferenceKey<RootTab>("rootTab", default: .home)
```

Comment:
Might not be needed now; but I could imagine it could also be expressed as a property wrapper?

Not perfect, but also quite expressive:
```suggestion
///     @LocalPreferenceKey("rootTab")
///     static let rootTab: RootTab = .home
```
or maybe the other way around?
```suggestion
///     @LocalPreferenceKey<RootTab>(default: .home)
///     static let rootTab = "rootTab"
```
Leaning toward the first version; we might also introduce it in future iterations.

But might also be confusing, as we would have two different property wrappers with similar names ...